src/combinatorics/hall/basic.lean

```diff
@@ -0,0 +1,250 @@
+/-
+Copyright (c) 2021 Alena Gusakov, Bhavik Mehta, Kyle Miller. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Alena Gusakov, Bhavik Mehta, Kyle Miller
+-/
+import combinatorics.hall.finite
+import topology.category.Top.limits
+
+/-!
+# Hall's Marriage Theorem
+
+Given a list of finite subsets $X_1,X_2,\dots,X_n$ of some given set
```

Comment:
This isn't changing anything for Latex, just makes readability ever so slightly better in text.
```suggestion
Given a list of finite subsets $X_1, X_2, \dots, X_n$ of some given set
```
Same below